File: LeetCode/53最大子数组和.java
class Solution {
    public int maxSubArray(int[] nums) {
        int cnt = 0, ans = Integer.MIN_VALUE;
        for (int i = 0; i < nums.length; i++) {
            cnt += nums[i];
            if (cnt > ans) ans = cnt;
            if (cnt <= 0) cnt = 0;
        }
        return ans;
    }
}

File: LeetCode/53最大子数组和.go
func maxSubArray(nums []int) (ans int) {
	dp := make([]int, len(nums))
	dp[0] = nums[0]
	ans = nums[0]
	for i := 1; i < len(nums); i++ {
		dp[i] = int(math.Max(float64(dp[i-1]+nums[i]), float64(nums[i])))
		if dp[i] > ans {
			ans = dp[i]
		}
	}
	return
}